src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java

```diff
@@ -237,13 +276,14 @@ private boolean initializeFolder(String folder) {
     }
 
     @Nullable
-    private synchronized Boolean compileItemsInternal() {
-        logInfo("Compiling items");
+    private synchronized Boolean compileItemsAndThingsInternal() {
+        logInfo("Compiling items and things");
         itemGenerator.generateItemsSource(itemRegistry.getItems());
+        thingGenerator.generateThingsSource(thingRegistry.getAll());
         Boolean result = false;
-        if (compiler.compileItems()) {
+        if (compiler.compileItemsAndThings()) {
 
-            logInfo("Creating items jar");
+            logInfo("Creating items and things jar");
             JRuleUtil.createJarFile(config.getItemsRootDirectory(),
                     compiler.getJarPath(JRuleCompiler.JAR_JRULE_ITEMS_NAME));
             result = true;
```

Comment:
Wouldn't it be better to either generate a separate JAR file or rename the current one?

Reply:
Yes, I think we should rename it, and not create separate files (with compiled actions we'll have 3 or more jar files to joggle)

Reply:
renamed to jrule-generated.jar